resolver/src/main/java/io/netty/resolver/RoundRobinInetAddressResolver.java

```diff
@@ -97,4 +97,17 @@ public void operationComplete(Future<List<InetAddress>> future) throws Exception
             }
         });
     }
+
+    /**
+     * @param anyInt any integer value to covert
+     * @param len max limit of range
+     * @return integer in (0, len] range, using formula {@code (anyInt % len + anyInt) % len}
+     */
+    protected static int toIndex(int anyInt, int len) {
+        int index = anyInt % len;
+        if (index < 0) {
+            index = index + len;
```

Comment:
> Your continuity will break from Integer.MAX_VALUE to Integer.MIN_VALUE

It's fine. You are trying to handle a gap which is a result of the absence of -Integer.MIN_VALUE. I think it's not a good idea, especially because it introduces a gap around 0. I'm pretty sure there is no easy way to do it without a gap (if we don't write non-obvious tricks that take into account turn number over Integer scale). Not worth any effort.